packages/node_modules/@webex/redux-module-meetings/src/actions.js

```diff
@@ -0,0 +1,208 @@
+import {buildDestinationLookup} from './reducer';
+
+export const JOIN_MEETING = 'meetings/JOIN_MEETING';
+export const LEAVE_MEETING = 'meetings/LEAVE_MEETING';
+export const STORE_MEETING = 'meetings/STORE_MEETING';
+export const UPDATE_MEETING_STATE = 'meetings/UPDATE_MEETING_STATE';
+
+// Redux Actions
+
+function joinMeeting(meeting) {
+  return {
+    type: JOIN_MEETING,
+    payload: {
+      meeting
+    }
+  };
+}
+
+function leaveMeetingAction(meeting) {
+  return {
+    type: LEAVE_MEETING,
+    payload: {
+      meeting
+    }
+  };
+}
+
+function storeMeeting({destinationType, destinationId, meeting}) {
+  return {
+    type: STORE_MEETING,
+    payload: {
+      destinationType,
+      destinationId,
+      meeting
+    }
+  };
+}
+
+function updateMeetingState(meeting, meetingState) {
+  return {
+    type: UPDATE_MEETING_STATE,
+    payload: {
+      meeting,
+      meetingState
+    }
+  };
+}
+
+// Helper Methods
+
+function bindMeetingEvents(meeting, dispatch) {
+  // Handle media streams changes to ready state
+  meeting.on('media:ready', (media) => {
+    if (!media) {
+      return;
+    }
+    if (media.type === 'local') {
+      dispatch(updateMeetingState(meeting, {hasLocalMedia: true}));
+    }
+    if (media.type === 'remoteVideo') {
+      dispatch(updateMeetingState(meeting, {hasRemoteVideo: true}));
+    }
+    if (media.type === 'remoteAudio') {
+      dispatch(updateMeetingState(meeting, {hasRemoteAudio: true}));
+    }
+  });
+
+  // TODO: Add media:stopped handlers
+
+  // TODO: Add meeting stopped & destroyed handlers
+}
+
+/**
+ * Gets the meeting instance from the SDK plugin
+ *
+ * @param {Object} config
+ * @param {String} config.destinationId
+ * @param {String} config.destinationType
+ * @param {String} config.meetingId
+ * @param {Object} config.meetings Redux state object from the store
+ * @param {Object} sdkInstance
+ * @returns {Object} meeting instance
+ */
+function getMeetingFromSDK({
+  destinationId,
+  destinationType,
+  meetingId,
+  meetings
+}, sdkInstance) {
+  let calculatedMeetingId = meetingId;
+
+  if (!meetingId) {
+    if (!destinationId || !destinationType || !meetings) {
+      throw new Error('#getMeetingFromSDK - unable to lookup meeting');
+    }
+
+    const destination = buildDestinationLookup({destinationId, destinationType});
+
+    calculatedMeetingId = meetings.getIn(['byDestination', destination]);
```

Comment:
@adamweeks  when would the developer use this function ?

Reply:
Since we are not passing around the meeting object or storing it in redux (as discussed with @chrisadubois) we need some way to get the meeting ID from the widget props of destination. It is used in the `addMediaToMeeting` and `leaveMeeting` action creator just below.